py/htm/advanced/regions/RawSensor.py

```diff
@@ -0,0 +1,169 @@
+# ----------------------------------------------------------------------
+# HTM Community Edition of NuPIC
+# Copyright (C) 2017, Numenta, Inc. 
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.    If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from collections import deque
+from htm.bindings.regions.PyRegion import PyRegion
+
+from .import extractList
+
+
+class RawSensor(PyRegion):
+    """
+    RawSensor is a simple sensor for sending sparse data into networks.
+
+    It accepts data using the command "addDataToQueue" or through the function
+    addDataToQueue() which can be called directly from Python. Data is queued up
+    in a FIFO and each call to compute pops the top element.
+
+    Each data record consists of the non-zero indices of the sparse vector,
+    a 0/1 reset flag, and an integer sequence ID.
```

Comment:
@dkeeney do we need this region? "sparse vector" is exactly our SDR, and I think the main InputRegion was adapted to accept SDRs? 
Q is if it supports the additional bells&whistles (0/1 reset flag, and sequence ID)?

Reply:
> @dkeeney do we need this region? "sparse vector" is exactly our SDR,

I think that this is not so much a facility to create a sparse vector but a facility to inject data into a link that can flow to other regions.    Similar to what VectorFileSensor does with data from a file.  I am thinking this might be implemented as a kind of encoder.

> " and I think the main InputRegion "

Not following you on this statement.